cirq/experiments/xeb_fitting.py

```diff
@@ -289,10 +306,209 @@ def _mean_infidelity(angles):
+            verbose=False,
+            pool=None,
+        )
+
+
+def characterize_phased_fsim_parameters_with_xeb_by_pair(
+    sampled_df: pd.DataFrame,
+    parameterized_circuits: List['cirq.Circuit'],
+    cycle_depths: Sequence[int],
+    phased_fsim_options: XEBPhasedFSimCharacterizationOptions,
+    initial_simplex_step_size: float = 0.1,
+    xatol: float = 1e-3,
+    fatol: float = 1e-3,
+    pool: Optional['multiprocessing.pool.Pool'] = None,
+) -> XEBCharacterizationResult:
+    """Run a classical optimization to fit phased fsim parameters to experimental data, and
+    thereby characterize PhasedFSim-like gates grouped by pairs.
+
+    This is appropriate if you have run parallel XEB on multiple pairs of qubits.
+
+    Args:
+        sampled_df: The DataFrame of sampled two-qubit probability distributions returned
+            from `sample_2q_xeb_circuits`.
+        parameterized_circuits: The circuits corresponding to those sampled in `sampled_df`,
+            but with some gates parameterized, likely by using `parameterize_phased_fsim_circuit`.
+        cycle_depths: The depths at which circuits were truncated.
+        phased_fsim_options: A set of options that controls the classical optimization loop
+            for characterizing the parameterized gates.
+        initial_simplex_step_size: Set the size of the initial simplex for Nelder-Mead.
+        xatol: The `xatol` argument for Nelder-Mead. This is the absolute error for convergence
+            in the parameters.
+        fatol: The `fatol` argument for Nelder-Mead. This is the absolute error for convergence
+            in the function evaluation.
+        pool: An optional multiprocessing pool to execute pair optimization in parallel. Each
+            optimization (and the simulations therein) runs serially.
+    """
+    pairs = sampled_df['pair'].unique()
+    closure = _CharacterizePhasedFsimParametersWithXebClosure(
+        parameterized_circuits=parameterized_circuits,
+        cycle_depths=cycle_depths,
+        phased_fsim_options=phased_fsim_options,
+        initial_simplex_step_size=initial_simplex_step_size,
+        xatol=xatol,
+        fatol=fatol,
+    )
+    subselected_dfs = [sampled_df[sampled_df['pair'] == pair] for pair in pairs]
+    if pool is not None:
+        results = pool.map(closure, subselected_dfs)
+    else:
+        results = [closure(df) for df in subselected_dfs]
+
+    optimization_results = {}
+    all_final_params = {}
+    fid_dfs = []
+    for result in results:
+        optimization_results.update(result.optimization_results)
+        all_final_params.update(result.final_params)
+        fid_dfs.append(result.fidelities_df)
+
+    return XEBCharacterizationResult(
+        optimization_results=optimization_results,
+        final_params=all_final_params,
+        fidelities_df=pd.concat(fid_dfs),
+    )
+
+
+def exponential_decay(cycle_depths: np.ndarray, A: float, layer_fid: float) -> np.ndarray:
```

Comment:
A?

Reply:
this is my go-to for the multiplicative constant in exponential functions: `y = A e^{b x}`. I don't know if there's a descriptive name that can be truly given. Theoretically, it should be representative of SPAM error. Do you have a suggestion for a better name?

Reply:
See, for example: https://en.wikipedia.org/wiki/Exponential_function

They don't capitalize it. 

In the context of fitting: https://mathworld.wolfram.com/LeastSquaresFittingExponential.html they do capitalize it

Reply:
It was very strange to see this variable name here, especially since it lights up as a constant because of Python syntax coloring, Maybe it could be named just scale_constant, scale_factor?

Btw, this is not my only comment on this PR, I need a little bit more time to read into this more though. I'll do it by Monday morning PT.

Reply:
would you be happier with a lowercase `a`? scale_factor could refer to scaling along the x-axis as well. I don't think we should try to prescribe too much meaning to a fit parameter that doesn't have a clear definition. See e.g. https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.curve_fit.html for more examples of people using a, b, c as fit parameters